cmake/options.cmake

```diff
@@ -33,6 +33,7 @@ else()
 endif()
 
 option(BUILD_TESTS          "Build tests"                                                   OFF)
+option(REGISTER_TESTS       "Register tests in CTest"                                       ON)
```

Comment:
Why is this extra cmake variable needed? The tests should be added when `BUILD_TESTS` are added?

Reply:
I just wanted to be able to align the targets with the Makefile ones by being able to disable these which do not exist there.

I wasn't even aware the CMake code registers tests. And it partially duplicates what `runtests.sh` is doing.

Reply:
What the reason to use the same targets as the makefile? Using ctest is superior at running the tests, and they can all be ran in parallel.

Reply:
Just to align it so you can use the same targets with the Makefile generated by CMake as the one we have. It's to align things and we need to clean up and deprecated things at a later date.

Reply:
The CTest things also miss a lot of things `runtest.sh` does - hence the failing build. Will take some time to align them to actually do the same.